hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java

```diff
@@ -289,6 +372,37 @@ private static boolean isAtLeastOneTagActive(final String[] tagNames) {
         };
     }
 
+    /**
+     * {@link Specification} for retrieving {@link Target}s that are in a given
+     * {@link RolloutGroup}
+     *
+     * @param group
+     *            the {@link RolloutGroup}
+     * @return the {@link Target} {@link Specification}
+     */
+    public static Specification<JpaTarget> isInRolloutGroup(final Long group) {
+        return (targetRoot, query, cb) -> {
+            final ListJoin<JpaTarget, RolloutTargetGroup> targetGroupJoin = targetRoot
+                    .join(JpaTarget_.rolloutTargetGroup);
+            return cb.equal(targetGroupJoin.get(RolloutTargetGroup_.rolloutGroup).get(JpaRolloutGroup_.id), group);
```

Comment:
could you please evaluate the performance of `isInRolloutGroup` and `isInActionRolloutGroup`, there seems to be a lot of joins happening, could you compare it with subquery approach?

Reply:
Previously these were resolved by a query derived from the method name, in case of `isInRolloutGroup` this was `findByRolloutTargetGroupRolloutGroupId`, which should result in an equivalent query.

The resulting query looks similar to this (shortened SQL debug output): `SELECT t1.* FROM sp_rollouttargetgroup t0, sp_target t1 WHERE (((t0.rolloutGroup_Id = ?) AND (t1.tenant = ?)) AND (t0.target_id = t1.id))` so only a single equi join, that is necessary to get the intended result, a subquery seems not necessary.

The situation for `isInActionRolloutGroup` is similar.

Reply:
ok, I just wanted to make sure we are not over-joining or eagerly loading the dependencies. I guess it builds the optimal query in this case due to the already present `id` value.